lib/notes.js

```diff
@@ -0,0 +1,77 @@
+'use strict';
+const Input = require('./input.js');
+const schemaModule = require('../models/note.schema.js');
+const mongoose = require('mongoose');
+
+class NoteHandler {
+  constructor(command) {
+    if (command && command.action) {
+      this.excute(command);
+
+    }
+    else {
+      console.error('Invalid Input');
+    }
+  }
+  excute(command) {
+    console.log('excute ', command);
+    switch (command.action) {
+    case 'add':
+      this.add(command.payload, command.category);
+      if (!command.payload) {
+        console.error('wrong flag');
+        return;
+      }
+
+      break;
+    case 'list':
+      this.list(command.payload, command.category);
+      break;
+    case 'delete':
+      if (!command.payload) {
+        console.log('there is no note to be deleted');
+        break;
+      }
+      this.delete(command.payload);
+      break;
+    default:
+      break;
+    }
+
+  }
+}
+
+// async add(text, category) {
```

Comment:
Looks like you have an older version of the add function here, but it's commented out and outside of the class brackets. I would replace most of the contents of this file with calling the `create` model function :) 

import your model at the top of the file: 
```
const Model = require('../models/notes-model.js')
```

and then call the create function on an instance of the model within the `add` function: 

```
let model = new Model(); 
model.create( /* your new record */ ); 
```